torch/csrc/jit/passes/replacement_of_old_operators.cpp

```diff
@@ -0,0 +1,89 @@
+#include <torch/csrc/jit/passes/replacement_of_old_operators.h>
+
+#include <c10/util/Exception.h>
+#include <caffe2/serialize/versions.h>
+#include <torch/csrc/jit/ir/irparser.h>
+#include <torch/csrc/jit/operator_upgraders/upgraders.h>
+#include <torch/csrc/jit/operator_upgraders/utils.h>
+#include <torch/csrc/jit/operator_upgraders/version_map.h>
+#include <torch/csrc/jit/runtime/graph_iterator.h>
+#include <limits>
+#include <regex>
+#include <string>
+
+namespace torch {
+namespace jit {
+
+struct OldOpsReplacer {
+  OldOpsReplacer(std::shared_ptr<Graph> graph) : graph_(std::move(graph)) {}
+
+  void run() {
+    if (!graph_->get_op_version().has_value()) {
+      return;
+    }
+    auto current_version = graph_->get_op_version().value();
+    DepthFirstGraphNodeIterator graph_it(graph_);
+    Node* node = graph_it.next();
+    int updated_version = caffe2::serialize::kMinSupportedFileFormatVersion;
+    while (node) {
+      if (auto schema = node->maybeSchema()) {
+        auto schema_name = schema->name() +
+            (schema->overload_name() != "" ? "." + schema->overload_name()
+                                           : "");
+        // this implies there was a version bump because of this operator
+        auto version_entry = get_operator_version_map().find(schema_name);
+        if (version_entry != get_operator_version_map().end()) {
+          const auto& entry = version_entry->second;
+          updated_version = std::max(
+              updated_version, entry[entry.size() - 1].bumped_at_version);
+          auto upgrader_entry =
+              findUpgrader(version_entry->second, current_version);
+          if (!upgrader_entry.has_value()) {
+            if (!isOpSymbolCurrent(schema_name, current_version)) {
+              TORCH_INTERNAL_ASSERT(
+                  false, "Upgrader must be present for ", schema_name);
+            }
+            return;
```

Comment:
Should this be `continue` after moving node to next?

Reply:
Yeah that is true. I was thinking if we see an up-to-date version of the operator, we shouldn't bother applying upgraders anymore. But yeah, it might not be the case for other operators. It should indeed be `continue`

Reply:
done